### PR TITLE
`executor`: 100% unit test coverage & fix dead blocks

### PR DESCRIPTION
**Pull request by chris-ramon**
#### Details
- `executor`: 100% unit test coverage & fix dead blocks
```
executor.go: 98.7% → 100.0% coverage
Lines	Issue	Fix
368-370	typeFromAST error check unreachable for *ast.Named + latent nil-ptr bug (conditionalType.Name() panics when type not in schema)	err != nil || conditionalType == nil
389-391	Same pattern in InlineFragment branch	err != nil || conditionalType == nil
497-499	Dead len(tOptions) == 0 — strings.Split never returns empty slice in Go	Removed the check
Tests added: TestDoesFragmentConditionMatch_FragmentDefinitionTypeNotFound and TestDoesFragmentConditionMatch_InlineFragmentTypeNotFound — both reference a type name not in the schema, hitting the new conditionalType == nil guard.
```

#### Test Plan
- Unit tests.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved fragment condition matching so unresolved/nullable derived type info no longer qualifies as a match.
  * Refined default resolver struct-field tag matching to rely solely on the primary tag option for field selection.
* **Tests**
  * Added extensive executor coverage for edge cases across planning/execution, directives, fragments, type resolution, error handling, subscription flows, and default resolver reflection behaviors (including tag and map access).

### CODE REVIEW

File: executor_internal_test.go [hunk 1]
```diff
@@ -0,0 +1,855 @@
...
+	if val != "value" {
+		t.Fatalf("expected 'value', got %v", val)
+	}
+}
+
+// ---------------------------------------------------------------------------
+// DefaultResolveFn – reflect.Map with func() interface{} value
+// ---------------------------------------------------------------------------
+
+type customMap map[string]interface{}
```
[the remark this answers is not in the document]
Comment (chris-ramon) [reply executor_internal_test.go]: Nice catch! — Addressed it via: 7304e7d

**Comment by chris-ramon** [comment]
LGTM :+1: :ship:
